### DISCUSSION TITLE
How do you send a Message to a Group?

### DISCUSSION THREAD

**Discussion by CairbreMadog** [bbernhard/signal-cli-rest-api#351] [Q&A]
I do not see any documentation on how to send a message to a group? It looks like v1 message sending API endpoint has/had a flag for "is group".

**Comment by bbernhard** [comment]
With the `groups` GET endpoint you can list all the existing groups. Then use the `id` in the `/v2/send` endpoint as recipient.

**Reply by CairbreMadog** [reply]
Thanks... honestly I somehow missed the "group." prepend and was trying to send to the ID without it. Thanks for replying.

**Reply by p6002** [reply]
> With the `groups` GET endpoint you can list all the existing groups. Then use the `id` in the `/v2/send` endpoint as recipient.

Is it possible to send msg to the group I currently belong to? I don't see it in the list "old" groups, I only see the currently created ones (by curl). 
Maybe in android is a way to preview the group.id?

**Comment by Dirk-Ingenieur** [comment]
I also have some trouble this group messages.

Here my examples:
1. Sending a message to a single number works fine:
[core-ssh ~]$ curl -X POST -H "Content-Type: application/json" 'http://192.168.1.205:8080/v2/send' -d '{"message": "Test via Signal API!", "number": "+49xxxxxxxxxxxxxxxxxx", "recipients": [ "+49yyyyyyyyyyyyyyyyy" ]}'

Reply is a timestamp: {"timestamp":"1716131739271"}

2. Sending a message to two recipients also works fine:
[core-ssh ~]$ curl -X POST -H "Content-Type: application/json" 'http://192.168.1.205:8080/v2/send' -d '{"message": "Test via Signal API!", "number": "+49xxxxxxxxxxxxxxxxxxx", "recipients": [ "+49yyyyyyyyyyyyyyyyy", "+49zzzzzzzzzzzzz" ]}'

3. So I created a group with both numbers:
 curl -X POST -H "Content-Type: application/json" -d '{"name": "Smarthome Admins", "members": ["+49yyyyyyyyyyy", "+49zzzzzzzzzzzz"]}' 'http://192.168.1.205:8080/v1/groups/+49xxxxxxxxxxxxxxx'
 group id was returned.
{"id":"group.RHpCZUE4xxxxxxxxfsaekdyTWlGOG9Wb3I2Z2lUQ0h6eC9JeldHUFBza1luWTIyTT0="}

4. I joined on both mobile phones the new group and sent messages just to check.
group communication works

5. Sending a message to the group
curl -X POST -H "Content-Type: application/json" 'http://192.168.1.205:8080/v2/send' -d '{"message": "Test via Signal API!", "number": "+49xxxxxxxxxx", "recipients": [ "group.RHpCZUE4xxxxxxxxfsaekdyTWlGOG9Wb3I2Z2lUQ0h6eC9JeldHUFBza1luWTIyTT0=" ]}'
This was replied with a timestamp as expected.

But for whatever reason the message cannot be seen in the group chat.

6. I tried debugging to figure out if something is wrong with the groups:
curl -X GET -H "Content-Type: application/json" 'http://192.168.1.205:8080/v1/groups/+49xxxxxxxxxx'

...
{"name":"Smarthome Admins","id":"group.dfkjfksakf3I2Z2lUQ0h6eC9JeldHUFBza1luWTIyTT0=","internal_id":"jsdfkjsdfkjagiTCHzx/IzWGPPskYnY22M=","members":["+49xxxxx"],"blocked":false,"pending_invites":[],"pending_requests":["+49yyyyyyyyyyyyyy","+49zzzzzzzz"],"invite_link":"","admins":["+49xxxxxxxxxxxxxxxx"]}
...
Does anybody have an idea whats wrong - what does pending request mean? Both mobiles accepted and joined the group.













5.

**Reply by louisdmueller** [reply]
I had the same error today, updating the container to the current version fixed it.
